data/tutorials/language/0lg_09_env.md

```diff
@@ -0,0 +1,675 @@
+
+Putting the parentheses the other way does not work:
+```ocaml
+# let bogus_cat : (string -> string) -> string = sweet_cat;;
+Error: This expression has type string -> string -> string
+       but an expression was expected of type (string -> string) -> string
+       Type string is not compatible with type string -> string
+```
+
+The type `(string -> string) -> string` denotes functions that take a function as a parameter, not a function as a result as `sweet_cat`.
+
+In computer science language, the type arrow operator _associates to the right_. Function types without parentheses should be treated as if they have parentheses to the right in the same way that the type of `dummy_cat` was declared above. Except they are not displayed. This is [pretty printing](https://en.wikipedia.org/wiki/Prettyprint).
+
+### Passing Tuples
+
+In OCaml, a *tuple* is a data structure used to group a fixed number of values, which can be of different types. Tuples are surrounded by parentheses, and the elements are separated by commas. Here's the basic syntax to create and work with tuples in OCaml:
+```ocaml
+# ("felix", 1920);;
+- : string * int = ("felix", 1920)
+```
+
+It is possible to use the tuple syntax to specify function parameters. Here is how it can be used to define yet another version of the running example:
+```ocaml
+# let spicy_cat (x, y) = x ^ " " ^ y;;
+val spicy_cat : string * string -> string = <fun>
+```
+
+It behaves the same as previously.
+```ocaml
+# spicy_cat ("hello", "world");;
+- : string = "hello world"
+```
+
+It looks like two arguments have been passed: `"hello"` and `"world"`. However, only one, the `("hello", "world")` pair, has been passed. Inspection of the generated assembly would show it isn't exactly the same function. It contains some more code. The contents of the pair passed to `spicy_cat` (`x` and `y`) must be extracted before evaluation of the `s ^ " " ^ t` expression. This is the role of the additional assembly instructions.
+
+In many imperative languages, the `spicy_cat ("hello", "world")` syntax reads as a function call with two parameters; but in OCaml, it denotes applying the function `spicy_cat` to a pair containing `"hello"` and `"world"`.
+
+### Currying and Uncurrying
+
+In the previous sections, two kinds of multiple-parameter functions have been presented.
+- Functions returning a function, such as `sweet_cat` and `sour_cat`
+- Functions taking a tuple as a parameter, such as `spicy_cat`
+
+Interestingly, both kinds of functions provide a way to pass several pieces of data to a body while being functions with a single parameter. From this perspective, it makes sense to say: “All functions have a single argument.”
```

Comment:
```suggestion
Interestingly, both kinds of functions provide a way to pass several pieces of data while being functions with a single parameter. From this perspective, it makes sense to say: “All functions have a single argument.”
```

What does "to a body" mean in this context? Does it work without it, as passing data (applying?) to functions is understood.

Reply:
Or remove those words like you suggest, it also works